includes/common.php

```diff
@@ -470,28 +431,8 @@ function getifdescrbyid($id)
 
 function getidbyname($hostname)
 {
-    global $cache;
-
-    if (isset($cache['devices']['hostname'][$hostname])) {
-        $id = $cache['devices']['hostname'][$hostname];
-    } else {
-        $id = dbFetchCell("SELECT `device_id` FROM `devices` WHERE `hostname` = ?", array($hostname));
-    }
-
-    return $id;
-}
-
-function gethostosbyid($id)
-{
-    global $cache;
-
-    if (isset($cache['devices']['id'][$id]['os'])) {
-        $os = $cache['devices']['id'][$id]['os'];
-    } else {
-        $os = dbFetchCell("SELECT `os` FROM `devices` WHERE `device_id` = ?", array($id));
-    }
-
-    return $os;
+    $device = DeviceCache::getByHostname($hostname);
+    return $device ? $device->device_id : null;
```

Comment:
This test is always true

Reply:
ah, yes, that was before I changed the cache to always return a device.